src/test/ui/consts/issue-66693-panic-in-array-len.rs

```diff
@@ -0,0 +1,17 @@
+// This is a separate test from `issue-66693.rs` because array lengths are evaluated
+// in a separate stage before `const`s and `statics` and so the error below is hit and
+// the compiler exits before generating errors for the others.
```

Comment:
> That's a good point, thanks. Maybe add a comment in the array-length-test-file explaining why this is a separate file.

@RalfJung already done.

Reply:
Okay. :) I'll review this when I find some more time.